### ISSUE TITLE
How to set compress_rate in Different conv ?

### ISSUE THREAD

**Issue by www516717402** [lmbxmu/HRankPlus#4]
Thank you for share this great project. Differ conv set differ compress rate in your paper, As follows. How to find this rate? If set fixed compress rate in all conv, have y compare with other SOTA prunning?
![image](https://user-images.githubusercontent.com/30460509/111961247-c92dab00-8b2b-11eb-9f50-033b490f0e43.png)

**Comment by lmbxmu** [comment]
> Thank you for share this great project. Differ conv set differ compress rate in your paper, As follows. How to find this rate? If set fixed compress rate in all conv, have y compare with other SOTA prunning?
> ![image](https://user-images.githubusercontent.com/30460509/111961247-c92dab00-8b2b-11eb-9f50-033b490f0e43.png)

Thanks for your interest. 
The pruning rates are empirically given in this paper.
A fixed compress rate does not affect the result much.
If you are interested, plz refer to https://github.com/lmbxmu/ABCPruner in which the pruning rates are automatically searched.

**Comment by www516717402** [comment]
Thank you for reply quickly. Assume compress rate not influence Net accuracy heavily,  When set all rate same as 0.8, this struct could refer to [paper](https://arxiv.org/abs/1810.05270) that only differ rate in structure or unstructure prunning will work. Have your try train net with non-load param after compress?
This [code](https://arxiv.org/abs/1810.05270) is help for me, thanks.

**Comment by lmbxmu** [comment]
> Thank you for reply quickly. Assume compress rate not influence Net accuracy heavily, When set all rate same as 0.8, this struct could refer to [paper](https://arxiv.org/abs/1810.05270) that only differ rate in structure or unstructure prunning will work. Have your try train net with non-load param after compress?
> This [code](https://arxiv.org/abs/1810.05270) is help for me, thanks.

Sorry. I made a mistake here. I was supposed to say that "A fixed compress rate does affect the result much."

Besides, I don't quite get your point of "train net with non-load param after compress."

**Comment by lmbxmu** [comment]
> Thank you for reply quickly. Assume compress rate not influence Net accuracy heavily, When set all rate same as 0.8, this struct could refer to [paper](https://arxiv.org/abs/1810.05270) that only differ rate in structure or unstructure prunning will work. Have your try train net with non-load param after compress?
> This [code](https://arxiv.org/abs/1810.05270) is help for me, thanks.

Sorry. I made a mistake here. I was supposed to say that "A fixed compress rate does affect the result much."

Besides, I don't quite get your point of "train net with non-load param after compress."

You can express what you wanna talk about in Chinese if you are from China.

**Comment by www516717402** [comment]
抱歉，英文表述有点问题。
1. 在剪枝完成之后，不加载预训练的模型（不使用彩票理论），直接重新初始化参数，这样训练的模型和论文相比如何？
2. 固定的压缩率没有意义（之前论文已经试验），论文中没有突出结构比参数更重要，以及如何确定不同层的剪枝比例。如果靠试验去搜索，和NAS基本没有区别，在工业上也很难应用。

**Comment by lmbxmu** [comment]
> 抱歉，英文表述有点问题。
> 
> 1. 在剪枝完成之后，不加载预训练的模型（不使用彩票理论），直接重新初始化参数，这样训练的模型和论文相比如何？
> 2. 固定的压缩率没有意义（之前论文已经试验），论文中没有突出结构比参数更重要，以及如何确定不同层的剪枝比例。如果靠试验去搜索，和NAS基本没有区别，在工业上也很难应用。

1. pre-trained的权重性能更好；这个有经验的人都知道
2. 我还是没明白你想说明什么； 再者学术也不全为工业服务

**Comment by www516717402** [comment]
> > 抱歉，英文表述有点问题。
> > 
> > 1. 在剪枝完成之后，不加载预训练的模型（不使用彩票理论），直接重新初始化参数，这样训练的模型和论文相比如何？
> > 2. 固定的压缩率没有意义（之前论文已经试验），论文中没有突出结构比参数更重要，以及如何确定不同层的剪枝比例。如果靠试验去搜索，和NAS基本没有区别，在工业上也很难应用。
> 
> 1. pre-trained的权重性能更好；这个有经验的人都知道
> 2. 我还是没明白你想说明什么； 再者学术也不全为工业服务

- [paper](https://arxiv.org/abs/1810.05270) pretrained权重在**正常操作下**都会提升性能和加快收敛，在剪枝小模型中（固定剪枝情况没有作用）不一定。
- 我想表达的意思是直接使用BN剪枝，设置不同层的比例，应该也能提高不少精度，关键点在于如何寻找这个比例。这个点在论文中没有体现。`The pruning rates are empirically given in this paper.` 

依然感谢解答疑惑。
